packages/paste-core/components/form-pill-group/src/FormPillGroup.tsx

```diff
@@ -39,26 +45,51 @@ export interface FormPillGroupProps
    * @memberof FormPillGroupProps
    */
   display?: "flex" | "inline-flex";
+  size?: FormPillGroupSizeVariant;
+  setPillFontSize: Dispatch<SetStateAction<FontSize | undefined>>;
 }
 
+/**
+ * Contains the style properties for the FormPillGroup component and the FormPill component.
+ */
+const SizeStyles: Record<FormPillGroupSizeVariant, Pick<BoxProps, "columnGap" | "rowGap" | "fontSize">> = {
+  default: {
+    columnGap: "space20",
+    rowGap: "space20",
+    fontSize: undefined,
+  },
+  large: {
+    columnGap: "space30",
+    rowGap: "space30",
+    fontSize: "fontSize30",
+  },
+};
+
 const FormPillGroupStyles = React.forwardRef<HTMLUListElement, FormPillGroupProps>(
-  ({ element = "FORM_PILL_GROUP", display = "flex", ...props }, ref) => (
-    <Box
-      {...safelySpreadBoxProps(props)}
-      element={element}
-      ref={ref}
-      role="listbox"
-      lineHeight="lineHeight30"
-      margin="space0"
-      padding="space0"
-      display={display}
-      flexWrap="wrap"
-      rowGap="space20"
-      columnGap="space20"
-    >
-      {props.children}
-    </Box>
-  ),
+  ({ element = "FORM_PILL_GROUP", display = "flex", size = "default", ...props }, ref) => {
+    const { fontSize, ...otherStyles } = SizeStyles[size];
+    useEffect(() => {
+      props.setPillFontSize(fontSize);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        element={element}
+        ref={ref}
+        role="listbox"
+        lineHeight="lineHeight30"
+        margin="space0"
+        padding="space0"
+        display={display}
+        flexWrap="wrap"
+        fontSize={fontSize}
+        {...otherStyles}
```

Comment:
suggestion(non-blocking): fontSize and can be spread like other styles. 

Could replace both lines with {...SizeStyles[size]}

Reply:
Yes. I can.